frontend/src/metabase/query_builder/components/view/ViewHeader/components/SavedQuestionLeftSide/util.ts

```diff
@@ -0,0 +1,21 @@
+import { useGetCardQueryMetadataQuery } from "metabase/api";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+import type { Table } from "metabase-types/api";
+
+export function useHiddenSourceTables(question: Question): Table[] {
+  const query = question.query();
+  const { isEditable, isNative } = Lib.queryDisplayInfo(query);
+
+  const canHaveHiddenSourceTables = !isEditable && !isNative;
+
+  const { data } = useGetCardQueryMetadataQuery(question.id(), {
+    skip: !canHaveHiddenSourceTables,
+  });
+
+  if (!canHaveHiddenSourceTables) {
+    return [];
+  }
+
+  return data?.tables.filter(table => table.visibility_type === "hidden") ?? [];
```

Comment:
This is incorrect - you need to check that `visibility_type != null`. There are visibility types which are all considered hidden. You can view them in Table metadata. Please make sure this is tested.